pkgs/misc/crust/default.nix

```diff
@@ -0,0 +1,113 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchpatch
+, bison
+, flex
+, buildPackages
+}:
+
+let
+  defaultVersion = "0.6";
+
+  defaultSrc = fetchFromGitHub {
+    owner = "crust-firmware";
+    repo = "crust";
+    rev = "v${defaultVersion}";
+    sha256 = "zalBVP9rI81XshcurxmvoCwkdlX3gMw5xuTVLOIymK4=";
+  };
+
+  buildCrust = {
+    version ? defaultVersion
+  , src ? defaultSrc
+  , defconfig
+  , extraConfig ? ""
+  , extraPatches ? []
+  , extraMakeFlags ? []
+  , extraMeta ? {}
+  , ... } @ args: stdenv.mkDerivation ({
+    pname = if (builtins.isNull ((builtins.match "_defconfig$" defconfig)))
+      then "crust-${lib.strings.removeSuffix "_defconfig" defconfig}"
+      else "crust-${defconfig}";
+
+    inherit version;
+    inherit src;
+
+    # Only for global crust patches, per-device paches should be included as extras to avoid adding noise which would complicate debugging
```

Comment:
Please wrap comments to 80 chars or so (not sure if there is an official column count, but these are too long).

Reply:
Also, typo: paches